docs/code/MlNetHighLevelConcepts.md

```diff
@@ -135,19 +137,49 @@ Of course, we can reduce this to the batch prediction:
 The above algorithm can be implemented using the [schema comprehension](SchemaComprehension.md), with two user-defined objects `InputExample` and `OutputPrediction` as follows:
 
 ```c#
-var inputData = env.CreateDataView(new InputExample[] { example });
+var inputData = mlContext.CreateDataView(new InputExample[] { example });
 var outputData = model.Transform(inputData);
-var output = outputData.AsDynamic.AsEnumerable<OutputPrediction>(env, reuseRowObject: false).Single();
+var output = outputData.AsDynamic.AsEnumerable<OutputPrediction>(mlContext, reuseRowObject: false).Single();
 ```
 
 But this would be cumbersome, and would incur performance costs. 
 Instead, we have a 'prediction function' object that performs the same work, but faster and more convenient, via an extension method `MakePredictionFunction`:
 
 ```c#
-var predictionFunc = model.MakePredictionFunction<InputExample, OutputPrediction>(env);
+var predictionFunc = model.MakePredictionFunction<InputExample, OutputPrediction>(mlContext);
 var output = predictionFunc.Predict(example);
 ```
 
 The same `predictionFunc` can (and should!) be used multiple times, thus amortizing the initial cost of `MakePredictionFunction` call. 
 
 The prediction function is *not re-entrant / thread-safe*: if you want to conduct predictions simultaneously with multiple threads, you need to have a prediction function per thread.
+
+## MLContext
+
+With a vast variety of components in ML.NET, it is important to have a consistent way to discover them. You can use the `MLContext` object for this purpose.
+
+Create one `MLContext` for your entire application. All ML.NET components and operations are available as methods of various sub-objects of `MLContext`.
+
+- `MLContext.Data` contains operations related to creating data views (other than transformers):
+  - Loading / saving data to a file
+  - Caching the data in memory
+  - Filtering (removing rows from data)
+
+Remember that data views are immutable, so each of the above operations create a new data view.
```

Comment:
This reads (to me) like a warning to not create too many data views. Is this the intent?

Reply:
No

---
In reply to: [233240748](https://github.com/dotnet/machinelearning/pull/1612#discussion_r233240748) [](ancestors = 233240748)